src/snowflake/connector/pandas_tools.py

```diff
@@ -444,10 +456,15 @@ def drop_object(name: str, object_type: str) -> None:
                 name=table_name,
                 quote_identifiers=quote_identifiers,
             )
-            drop_object(original_table_location, "table")
-            rename_table_sql = f"ALTER TABLE {target_table_location} RENAME TO {original_table_location} /* Python:snowflake.connector.pandas_tools.write_pandas() */"
-            logger.debug(f"rename table with '{rename_table_sql}'")
-            cursor.execute(rename_table_sql, _is_internal=True)
+            clone_table_sql = (
+                f"CREATE OR REPLACE TABLE {original_table_location} "
+                f"CLONE {target_table_location} "
+                f"{'COPY GRANTS' if copy_grants else ''}"
+                f" /* Python:snowflake.connector.pandas_tools.write_pandas() */ "
+            )
+            logger.debug(f"clone table with '{clone_table_sql}'")
+            cursor.execute(clone_table_sql, _is_internal=True)
+            drop_object(target_table_location, "table")
```

Comment:
this could be a breaking change since not all users my have create table privileges. Can you refactor the code like so

```
if clone_table_sql:
    clone_table_sql = "create or replace ..."
    logger.debug()
    cursor.execute()
    drop_object()
else:
    rename_table_sql = "alter table ..."
    logger.debug()
    cursor.execute()
```